### PR TITLE
AF-2294: Online Modeler MVP

### PR DESCRIPTION
**Pull request by paulovmr**
Relates to:
* https://github.com/kiegroup/appformer/pull/864
* https://github.com/kiegroup/kie-wb-common/pull/3062

### CODE REVIEW

File: packages/chrome-extension-pack-kogito-kie-editors/src/OnlineEditorManager.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+      response => {
+        if (!response?.success) {
+          console.debug("Error during online editor opening.");
+        }
+      }
+    );
+  }
+
+  public listenToComeBack(setFileName: (fileName: string) => void, setFileContent: (fileContent: string) => void) {
+    const listener = (request: any, sender: any, sendResponse: any) => {
```
Comment (tiagobento) [inline packages/chrome-extension-pack-kogito-kie-editors/src/OnlineEditorManager.ts]: Can these types not be `any`?

File: packages/chrome-extension-pack-kogito-kie-editors/src/background.ts [hunk 2]
```diff
@@ -38,3 +38,125 @@ chrome.webRequest.onHeadersReceived.addListener(
...
+      if (response?.success) {
+        chrome.tabs.remove(sender.tab!.id!, () => {
+          chrome.tabs.update(request.senderTabId, { active: true, selected: true });
+        });
+      }
+    }
+  );
+}
+
+function extractFileExtension(fileName: string) {
```
Comment (tiagobento) [inline packages/chrome-extension-pack-kogito-kie-editors/src/background.ts]: Can you put a FIXME here indicating that this code is duplicated? We should address that some time.

File: packages/chrome-extension-pack-kogito-kie-editors/src/online-editor-content-script.ts [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+  if (request.messageId === "LOAD_ONLINE_EDITOR") {
+    const event = new CustomEvent("loadOnlineEditor", {
+      detail: {
+        fileName: request.filePath,
+        fileContent: request.fileContent,
+        readonly: request.readonly,
+        senderTabId: request.senderTabId
+      }
+    });
+    window.dispatchEvent(event);
```
Comment (tiagobento) [inline packages/chrome-extension-pack-kogito-kie-editors/src/online-editor-content-script.ts]: I guess inlining this variable improves readability.

File: packages/chrome-extension/src/app/components/single/singleEditorView.tsx [hunk 4]
```diff
@@ -96,6 +97,7 @@ function SingleEditorViewApp(props: { fileInfo: FileInfo; openFileExtension: str
     <SingleEditorApp
       readonly={true}
       openFileExtension={props.openFileExtension}
+      fileName={decodeURIComponent(props.fileInfo.path.split("/").pop()!)}
```
Comment (tiagobento) [inline packages/chrome-extension/src/app/components/single/singleEditorView.tsx]: Can you please extract a variable? This line look like too much code for being inside the template.

File: packages/online-editor/src/common/Routes.ts [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+    }
+
+    const parsedEditorType = editorTypeRegex.pop();
+    if (!parsedEditorType) {
+      return undefined;
+    }
+
+    return parsedEditorType;
+  }
+}
```
Comment (tiagobento) [inline packages/online-editor/src/common/Routes.ts]: Good candidate for optional chaining?

File: packages/online-editor/src/editor/EditorPage.tsx [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+  onFileNameChanged: (fileName: string) => void;
+}
+
+enum ActionType {
+  NONE,
+  SAVE,
+  DOWNLOAD
+}
+
+let action = ActionType.NONE;
```
Comment (tiagobento) [inline packages/online-editor/src/editor/EditorPage.tsx:39]: Can you please put a FIXME here indicating that this is not ideal? We should address it some time.

**Comment by paulovmr** [comment]
@tiagobento All suggestions were implemented. Thanks!

**Review by tiagobento — APPROVED** [review]

**Pull request merged**